tests/tt_metal/tt_metal/unit_tests_fast_dispatch/common/command_queue_fixture.hpp

```diff
@@ -135,3 +152,291 @@ class SingleDeviceTraceFixture: public ::testing::Test {
     }
 
 };
+
+class RandomProgramFixture : public CommandQueueSingleCardFixture {
```

Comment:
I would prefer a set of helper functions rather than a Fixture. But not opposed to this

Reply:
I decided to create a separate fixture for these tests because the helper functions that I've added are only used in the randomized Program tests, so it makes sense for them to all be encapsulated together in the same fixture.